IHP/FrameworkConfig.hs

```diff
@@ -19,18 +18,26 @@ import IHP.Mail.Types
 defaultPort :: Int
 defaultPort = 8000
 
-portRef :: IORef Int
-portRef = unsafePerformIO (newIORef defaultPort)
+defaultFrameworkConfig :: Text -> Environment -> IO FrameworkConfig
+defaultFrameworkConfig appHostname environment = do
+    appPort <- defaultAppPort
+    databaseUrl <- defaultDatabaseUrl
+    let
+        baseUrl = let port = appPort in "http://" <> appHostname <> (if port /= 80 then ":" <> tshow port else "")
+        requestLoggerMiddleware = RequestLogger.logStdoutDev
+        sessionCookie = defaultIHPSessionCookie baseUrl
```

Comment:
I think there's an issue here. The `baseUrl` is passed here to check whether the session cookie needs to be `Secure` (meaning only available for https connections). Previously with the class based approach the `baseUrl` from the `FrameworkConfig` instance was passed.

An example: Let's say I'm running an IHP app on `https://example.com`. In the old `FrameworkConfig` I override the `baseUrl` to be `https://example.com`. Then the sessionCookie will be secure because `https://example.com` is passed to the `defaultIHPSessionCookie`.


But now with this change it would pass `let port = appPort in "http://" <> appHostname <> (if port /= 80 then ":" <> tshow port else "")` to the `defaultIHPSessionCookie` instead of the overriden `https://example.com`. Then the cookie would not be secure unless I override the `sessionCookie` function as well.

Reply:
Basically what we're missing is late binding. Will have to think a little bit more about how we can easily solve this here.

Reply:
Maybe just expand defaultFrameworkConfig into developer-land?